openspoor/visualisations/trackmap.py

```diff
@@ -17,60 +19,68 @@ class PlotObject(ABC):
     A parent class for every object that can be plotted on a TrackMap
     """
 
+    def __init__(self, data, popup):
+        if popup is None or isinstance(popup, list):
+            self.popup = popup
+        elif isinstance(popup, str):
+            self.popup = [popup]
+        else:
+            TypeError('Unknown type for popup in linestrings')
+
+        if isinstance(data, gpd.GeoDataFrame):
+            self.data = data
+        elif isinstance(data, str):
+            self.data = gpd.read_file(data)
+        else:
+            raise TypeError('Provide either a geopandas dataframe or a file location of a csv to show')
+        self.data = self.data.to_crs('EPSG:4326')
+        if self.popup:
+            self.data = self.data.set_index(self.popup)
+
     @abstractmethod
     def add_to(self, m) -> None:
         """
         A base function that should be overwritten with logic for every plottable object.
+
         :param m: A TrackMap object, to which this element should be added.
         :return: None
         """
         raise NotImplementedError('This needs to be set in implementation classes')
 
-    def _convert_pandas_to_geopandas(self, df):
-        # If this is a pandas dataframe with Rijksdriehoek in it, convert it to GPS
-        first_coord = df.geometry.values[0].split('(')[1].split(' ')[0]
-        if float(first_coord) > 180:
-            expected_crs = 'EPSG:28992'
```

Comment:
This part is removed now, does it still work for first_coords > 180?

Reply:
This part was an ugly fix to look for the CRS of the data; for Rijksdriehoek this should never happen, while for GPS it should always happen. As the data goes in as a geopandas geodataframe now (except for some cases for the plottingpoints, which do allow some pandas dataframes still) this functionality should not be required or used within this repo. Whenever users use lat and lon that should be in GPS; whenever they use geodataframes the conversion is fixed by geopandas. Therefore I felt that this could be safely removed.